src/msg.c

```diff
@@ -1364,6 +1315,32 @@ static bool handle_device_removed(const char *device_id)
 	return false;
 }
 
+static bool handle_schema_updated(struct session *session,
+				  const char *device_id, const char *err)
+{
+	struct knot_device *device;
+
+	if (err) {
+		hal_log_error("%s", err);
+		return true;
+	}
+
+	/* TODO: Send KNOT_MSG_SCHM_END_RSP using node_ops */
+
+	device = device_get(device_id);
+	if (device)
+		device_set_registered(device, true);
+
+	/*
+	 * For security reason, remove from rollback avoiding clonning attack.
+	 * If schema is being sent means that credentals (UUID/token) has been
+	 * properly received (registration complete).
+	 */
+	session->rollback = 0; /* Rollback disabled */
```

Comment:
Please remove the session->rollback in `handle_device_added`. The rollback was disabled there with purposes of tests only but the rollback should be disabled just here.

Reply:
Done.